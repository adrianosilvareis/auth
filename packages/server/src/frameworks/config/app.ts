import express from 'express'

import { routes as setupRoutes } from './routes'

export const app = express()

setupRoutes(app)
