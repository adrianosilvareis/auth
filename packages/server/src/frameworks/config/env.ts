let DB_NAME, DB_HOST
const NODE_ENV = process.env.NODE_ENV ?? 'development'

if (NODE_ENV === 'production') { DB_NAME = 'clean_node_api' } else if (NODE_ENV === 'development') { DB_NAME = 'cna_dev' } else { DB_NAME = 'cna_test' }
if (NODE_ENV === 'test') { DB_HOST = 'localhost' } else { DB_HOST = 'postgres' }

export const env = {
  mongoUrl: process.env.MONGO_URL ?? 'mongodb://mongo:27017/clean-node-api',
  postgresUrl: process.env.DATABASE_URL ?? `postgres://postgres:postgres@${DB_HOST}:5432/${DB_NAME}`,
  port: process.env.PORT ?? 3000,
  jwtSecret: process.env.JWT_SECRET ?? 'a2611p@r19'
}
