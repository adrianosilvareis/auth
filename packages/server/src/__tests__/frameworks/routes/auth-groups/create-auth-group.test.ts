import request from 'supertest'
import { app } from '@/frameworks/config'
import { prismaMock } from '@/__tests__/frameworks/database/prisma-client-mock'

describe('POST /api/auth-groups', () => {
  it('should return HttpResponse serverError on repository throws', async () => {
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    prismaMock.as_auth_groups.create.mockRejectedValueOnce(new Error('any_repository_error'))

    const { body, statusCode } = await request(app)
      .post('/api/auth-groups')
      .send({
        title: 'admin',
        activities: [
          {
            id: 'd0d8a425-7665-417d-9f70-0712c4bdae42',
            name: 'auth-group',
            permissions: [
              'readonly',
              'create',
              'delete',
              'update'
            ]
          }
        ]
      })

    expect(statusCode).toBe(500)
    expect(body).toMatchSnapshot()
  })
  it('should return HttpResponse Ok on success', async () => {
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    prismaMock.as_auth_groups.create.mockReturnValueOnce(mockAuthGroupPrismaRepository)

    const { body, statusCode } = await request(app)
      .post('/api/auth-groups')
      .send({
        title: 'admin',
        activities: [
          {
            id: 'd0d8a425-7665-417d-9f70-0712c4bdae42',
            name: 'auth-group',
            permissions: [
              'readonly',
              'create',
              'delete',
              'update'
            ]
          }
        ]
      })

    expect(statusCode).toBe(200)
    expect(body).toMatchSnapshot()
  })
})

const mockAuthGroupPrismaRepository = [
  {
    id: '37f2e75e-4b86-4a99-977e-2aea6d4c782b',
    title: 'Administrator',
    activities: [
      {
        activity_id: 'd0d8a425-7665-417d-9f70-0712c4bdae42',
        auth_groups_id: '37f2e75e-4b86-4a99-977e-2aea6d4c782b',
        createdAt: '2021-05-07T17:17:30.941Z',
        activity: {
          id: 'd0d8a425-7665-417d-9f70-0712c4bdae42',
          name: 'auth-group',
          permissions: ['readonly', 'create', 'delete', 'update']
        }
      }
    ]
  }
]
