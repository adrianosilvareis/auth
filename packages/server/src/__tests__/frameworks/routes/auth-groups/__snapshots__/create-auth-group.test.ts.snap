// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST /api/auth-groups should return HttpResponse Ok on success 1`] = `Object {}`;

exports[`POST /api/auth-groups should return HttpResponse serverError on repository throws 1`] = `
Object {
  "error": Object {
    "message": "Internal server error",
    "stack": "Error: any_repository_error
    at Object.<anonymous> (D:\\\\BIBLIOTECAS\\\\Desktop\\\\auth-monorepo\\\\packages\\\\server\\\\src\\\\__tests__\\\\frameworks\\\\routes\\\\auth-groups\\\\create-auth-group.test.ts:10:7)
    at Object.asyncJestTest (D:\\\\BIBLIOTECAS\\\\Desktop\\\\auth-monorepo\\\\node_modules\\\\@jest\\\\core\\\\node_modules\\\\jest-jasmine2\\\\build\\\\jasmineAsyncInstall.js:106:37)
    at D:\\\\BIBLIOTECAS\\\\Desktop\\\\auth-monorepo\\\\node_modules\\\\@jest\\\\core\\\\node_modules\\\\jest-jasmine2\\\\build\\\\queueRunner.js:45:12
    at new Promise (<anonymous>)
    at mapper (D:\\\\BIBLIOTECAS\\\\Desktop\\\\auth-monorepo\\\\node_modules\\\\@jest\\\\core\\\\node_modules\\\\jest-jasmine2\\\\build\\\\queueRunner.js:28:19)
    at D:\\\\BIBLIOTECAS\\\\Desktop\\\\auth-monorepo\\\\node_modules\\\\@jest\\\\core\\\\node_modules\\\\jest-jasmine2\\\\build\\\\queueRunner.js:75:41
    at processTicksAndRejections (node:internal/process/task_queues:94:5)",
  },
}
`;
